util/include/json_output.hpp

```diff
@@ -78,4 +80,56 @@ inline std::pair<std::string, std::string> toPair(const std::string &key, const
   return std::make_pair(key, value);
 }
 
+// Build a JSON object.  Supports nested objects.
+class BuildJson {
+ public:
+  inline BuildJson() {}
+
+  // Start a root JSON object.
+  inline void startJson() { str += "\n{\n"; }
```

Comment:
c++ ISO standard:
`A function defined within a class definition is an inline function.`

I think that all the explicit inline keywords are unneeded in a c++17 modern compiler.
The compiler knows it is an inline function since you've declared and defined it in the class declaration (header).
same for all functions below.